src/server.js

```diff
@@ -27,22 +27,87 @@ app.get('/', (req, res) => {
 });
 
 app.post('/api/chat', checkIsAuthenticated, async (req, res) => {
-  const { message } = req.body;
+  const { message, user_id, conversation_id } = req.body;
 
   if (!message) {
     return res.status(400).json({ error: "Message is required." });
   }
 
+  if (!user_id) {
+    return res.status(400).json({ error: "User ID is required." });
+  }
+
   try {
     const openaiResponse = await openaiRequest(message);
-    res.json({ response: openaiResponse });
+
+    let associatedConversationID;
+    let continuedConversation;
+    //if no conversation ID present, add a new conversation and use its ID - otherwise use the ID given
+    if (!conversation_id) {
+      continuedConversation = false;
+      const openaiSummary = await openaiRequest("Summarize the specific topic of the question that was asked in 5 words or less.");
+      //is there a way to achieve this without requesting a second time?
+      const newConversation = {
+        title: openaiSummary,
+        updated_at: new Date()
+      };
+
+      const newlyCreated = await addConversation(newConversation);
+      associatedConversationID = newlyCreated[0].id;
+    } else {
+      continuedConversation = true;
+      associatedConversationID = conversation_id;
+    }
+
+    const chatUser = await getChatUserByID(user_id);
+    const chatUsername = chatUser.username;
+
+    if (associatedConversationID && chatUsername) { //to make sure we have both before proceeding
+      //Preparing the message record for the question the user asked
+      const userQuestion = {
+        chat_user_id: user_id,
+        conversation_id: associatedConversationID,
+        author: chatUsername,
+        content: message
+      };
+
+      //Preparing the message record for the answer received from ChatGPT
+      const gptAnswer = {
+        chat_user_id: user_id,
+        conversation_id: associatedConversationID,
+        author: "ChatGPT", //Is this text ok?
```

Comment:
It seems that there are 3 roles in the OpenAI:
'user', 'assistant' and 'system'
Probably better to stick to those.